saleor/dashboard/category/views.py

```diff
@@ -0,0 +1,73 @@
+from django.contrib import messages
+from django.core.urlresolvers import reverse_lazy
+from django.shortcuts import get_object_or_404, redirect
+from django.template.response import TemplateResponse
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import DeleteView
+
+from ...product.models import Category
+from ..views import StaffMemberOnlyMixin, staff_member_required
+from .forms import CategoryForm
+
+
+@staff_member_required
+def category_list(request, root=None):
+    ctx = {}
+    categories = Category.objects.all()
+    if root:
+        current_node = get_object_or_404(Category, pk=root)
+        categories = current_node.get_descendants()
+        ctx['current_node'] = current_node
+        ctx['category_breadcrumbs'] = current_node.get_ancestors(include_self=True)
+    min_level = categories[0].get_level() if categories else 0
+    max_level = min_level + 1
+    categories = categories.filter(level__gte=min_level, level__lte=max_level)
+    ctx['categories'] = categories
+    ctx['min_level'] = min_level
+    return TemplateResponse(request, 'dashboard/category/list.html', ctx)
+
+
+@staff_member_required
+def category_details(request, pk=None, parent_pk=None):
+    ctx = {}
+    if pk:
+        category = get_object_or_404(Category.objects.all(), pk=pk)
+        title = category.name
```

Comment:
Isn't this the template's responsibility to handle? Changing UI strings should not require me to alter backend code.